src/stdio_mgr/stdio_mgr.py

```diff
@@ -181,41 +222,35 @@ def stdio_mgr(in_str=""):
 
     out_
 
-        :cls:`~io.StringIO` -- Temporary stream for `stdout`,
+        :cls:`~io.RandomTextIO` -- Temporary stream for `stdout`,
         initially empty.
 
     err_
 
-        :cls:`~io.StringIO` -- Temporary stream for `stderr`,
+        :cls:`~io.RandomTextIO` -- Temporary stream for `stderr`,
         initially empty.
 
     """
     old_stdin = sys.stdin
     old_stdout = sys.stdout
     old_stderr = sys.stderr
 
-    new_stdout = StringIO()
-    new_stderr = StringIO()
-    new_stdin = TeeStdin(new_stdout, in_str)
+    with ExitStack() as stack:
+        new_stdout = stack.enter_context(RandomTextIO())
+        new_stderr = stack.enter_context(RandomTextIO())
+        new_stdin = stack.enter_context(TeeStdin(new_stdout, in_str))
 
-    sys.stdin = new_stdin
-    sys.stdout = new_stdout
-    sys.stderr = new_stderr
+        close_files = stack.pop_all().close
```

Comment:
Ahhhh, this is binding the close *method* for (optional) later use; not invoking it. Nice.

Once all of these are `.pop_all()`-ed over to the new `ExitStack` instance, is there any reason to stay in the original `ExitStack` context?

Reply:
dedenting the rest is better. thx for spotting that.